### PR TITLE
Corrections in `alt_az.hpp` and `base_frame.hpp`

### PR DESCRIPTION
**Pull request by lpranam**
### Description
Header file names corrected in `alt_az.hpp`
missing `typename`s added
`get_angular_separation` function corrected in `base_frame.hpp`

### Tasklist

- [x] Ensure all CI builds pass
- [x] Review and approve

### CODE REVIEW

**Review by sarthak2007 — CHANGES_REQUESTED** [review]
add `typename` also in lines 96-101

**Review by sarthak2007 — CHANGES_REQUESTED** [review]
I guess `magnitude` function used is giving error as when it is defined in `arithmetic.hpp`, then the coordinate is not converted to `cartesian` but `get_x()` is used

**Review by Ruchika2696 — APPROVED** [review]

**Comment by lpranam** [comment]
@sarthak2007 that error occurred due to something went wrong in travis. Though where exactly you think the error is in `magnitude`?

**Comment by sarthak2007** [comment]
> @sarthak2007 that error occurred due to something went wrong in travis. Though where exactly you think the error is in `magnitude`?

Magnitude function used is giving error as when it is defined in `arithmetic.hpp`, then the coordinate(that is its argument) is not converted to cartesian but then also get_x() is used

File: include/boost/astronomy/coordinate/base_frame.hpp [hunk 1]
```diff
@@ -96,10 +96,11 @@ struct base_frame
...
 
     //!angular separation between two coordinates in radians
-    bu::quantity<bu::si::dimensionless> separation(base_frame const& other) const
+    bu::quantity<bu::si::plane_angle> get_angular_separation(base_frame const& other) const
     {
-        return bu::quantity<bu::si::dimensionless>::from_value(
-            std::acos(dot(this->data, other.get_data()))
+        return bu::quantity<bu::si::plane_angle>::from_value(
+            std::acos((dot(this->data, other.get_data()) / 
+                (magnitude(this->data) * magnitude(other))).value())
```
Comment (sarthak2007) [inline include/boost/astronomy/coordinate/base_frame.hpp]: Use `other.get_data()` instead of `other`

**Review by sarthak2007 — CHANGES_REQUESTED** [review]

**Review by sarthak2007 — APPROVED** [review]

**Pull request merged**